### PR TITLE
feat: [CI-7718]: Support drone-buildx plugin

### PR DESCRIPTION
**Pull request by rutvijmehta-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46312)

### CODE REVIEW

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/CIExecutionConfigService.java [hunk 1]
```diff
@@ -664,6 +664,8 @@ public String getContainerlessPluginNameForVM(CIStepInfoType stepInfoType) {
         name = vmContainerlessStepConfig.getGitCloneConfig().getName();
         break;
       case DOCKER:
+        name = vmContainerlessStepConfig.getDockerBuildxConfig().getName();
```
Comment (rutvijmehta-harness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/CIExecutionConfigService.java:667]: I'll add a FF to use `drone-buildx` binary for `Build and Push to Docker` step.

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/buildstate/PluginSettingUtils.java [hunk 2]
```diff
@@ -554,6 +556,16 @@ private static Map<String, String> getDockerStepInfoEnvVariables(
       }
     } else if (infraType == Type.VM) {
       setMandatoryEnvironmentVariable(map, PLUGIN_DAEMON_OFF, "true");
+      List<String> cacheFromList =
```
Comment (rutvijmehta-harness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/buildstate/PluginSettingUtils.java:559]: Check for FF here as well.

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/CIExecutionConfigService.java (hunk 1 above)
Comment (rutvijmehta-harness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/CIExecutionConfigService.java:667]: Done

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/buildstate/PluginSettingUtils.java (hunk 2 above)
Comment (rutvijmehta-harness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/buildstate/PluginSettingUtils.java]: Done

**Review by shubham149 — DISMISSED** [review]

**Review by shubham149 — DISMISSED** [review]

**Pull request merged**